packages/cli/src/sso/saml/routes/saml.controller.ee.ts

```diff
@@ -1,41 +1,50 @@
 import express from 'express';
+import { Get, Post, RestController } from '../../../decorators';
+import { SamlUrls } from '../constants';
 import {
 	samlLicensedAndEnabledMiddleware,
 	samlLicensedOwnerMiddleware,
 } from '../middleware/samlEnabledMiddleware';
 import { SamlService } from '../saml.service.ee';
-import { SamlUrls } from '../constants';
-import type { SamlConfiguration } from '../types/requests';
-import { AuthError, BadRequestError } from '@/ResponseHelper';
+import { SamlConfiguration } from '../types/requests';
+import { AuthError, BadRequestError } from '../../../ResponseHelper';
+import { getInitSSOFormView } from '../views/initSsoPost';
+import { getInitSSOPostView } from '../views/initSsoRedirect';
 import { issueCookie } from '../../../auth/jwt';
 import { validate } from 'class-validator';
 import type { PostBindingContext } from 'samlify/types/src/entity';
-import { getInitSSOFormView } from '../views/initSsoPost';
-import { getInitSSOPostView } from '../views/initSsoRedirect';
 
-export const samlControllerProtected = express.Router();
+@RestController('/sso/saml')
+export class SamlController {
+	private samlService: SamlService;
 
-/**
- * GET /sso/saml/config
- * Return SAML config
- */
-samlControllerProtected.get(
-	SamlUrls.config,
-	samlLicensedOwnerMiddleware,
-	(req: SamlConfiguration.Read, res: express.Response) => {
+	constructor() {
+		this.samlService = SamlService.getInstance();
+	}
```

Comment:
Please change this to 
```suggestion
	constructor(private samlService: SamlService) {}
```

and pass the service instance from the constructor call. That makes mocking a lot simpler.

Reply:
Done